examples/nanocoap_tool/main.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2024 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       Example application to access RIOT shell commands from Linux
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include "shell.h"
+#include "ztimer.h"
+
+int main(int argc, char **argv)
+{
+    if (argc < 1) {
+        printf("usage: <command> [args]\n");
+        return 0;
+    }
+
+    shell_command_handler_t handler = shell_find_handler(NULL, argv[0]);
```

Comment:
I think this is fine as it but it does break with the usual convention that `argv[0]` contains the application name. Do you think a code comment could prevent confusion?